LinuxSerialDriver and Tcp Driver Read Threads Need to Switch to Unsigned Types

### Discussed in https://github.com/nasa/fprime/discussions/1343

<div type='discussions-op-text'>

<sup>Originally posted by **danjwait** March 22, 2022</sup>
Trying to complete the GPS tutorial I ran into this during `fprime-util build` of /GpsApp :
```
...
Scanning dependencies of target GpsApp
[100%] Building CXX object CMakeFiles/GpsApp.dir/Top/Main.cpp.o
In file included from /home/djwait/02_Projects/fprime/Drv/LinuxSerialDriver/LinuxSerialDriver.hpp:9,
                 from /home/djwait/02_Projects/fprime/GpsApp/build-fprime-automatic-native/GpsApp/Top/GpsAppTopologyAc.hpp:16,
                 from /home/djwait/02_Projects/fprime/GpsApp/Top/Main.cpp:6:
/home/djwait/02_Projects/fprime/Drv/LinuxSerialDriver/LinuxSerialDriverComponentImpl.hpp:74:65: error: conversion to ‘NATIVE_INT_TYPE’ {aka ‘int’} from ‘NATIVE_UINT_TYPE’ {aka ‘unsigned int’} may change the sign of the result [-Werror=sign-conversion]
   74 |       void startReadThread(NATIVE_INT_TYPE priority = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE stackSize = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE cpuAffinity = Os::Task::TASK_DEFAULT);
      |        
...
```
There were three of the same error, all on the same line (74):
```
...
 //! start the serial poll thread.
      //! buffSize is the max receive buffer size
      //!
      void startReadThread(NATIVE_INT_TYPE priority = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE stackSize = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE cpuAffinity = Os::Task::TASK_DEFAULT);
...
```
 I see that the three Os::Task::TASK_DEFAULT are NATIVE_UINT_TYPE , so I worked around that by changing all the `NATIVE_INT_TYPE` to `NATIVE_UINT_TYPE` ; not sure if that was the right thing to do, but the GpsApp was able to complete building.

Should I have done something else? Did I goof up something else?</div>

--- a/Drv/Ip/SocketReadTask.cpp
+++ b/Drv/Ip/SocketReadTask.cpp
@@ -25,9 +25,9 @@ SocketReadTask::~SocketReadTask() {}
 
 void SocketReadTask::startSocketTask(const Fw::StringBase &name,
                                      const bool reconnect,
-                                     const NATIVE_INT_TYPE priority,
-                                     const NATIVE_INT_TYPE stack,
-                                     const NATIVE_INT_TYPE cpuAffinity) {
+                                     const NATIVE_UINT_TYPE priority,
+                                     const NATIVE_UINT_TYPE stack,
+                                     const NATIVE_UINT_TYPE cpuAffinity) {
     FW_ASSERT(not m_task.isStarted());  // It is a coding error to start this task multiple times
     FW_ASSERT(not this->m_stop);        // It is a coding error to stop the thread before it is started
     m_reconnect = reconnect;

--- a/Drv/Ip/SocketReadTask.hpp
+++ b/Drv/Ip/SocketReadTask.hpp
@@ -41,19 +41,19 @@ class SocketReadTask {
      *
      * Starts up the socket reading task and opens socket. This should be called before send calls are expected to
      * work. Will connect to the previously configured port and host. priority, stack, and cpuAffinity are provided
-     * to the Os::Task::start call. cpuAffinity defaults to -1.
+     * to the Os::Task::start call.
      *
      * \param name: name of the task
      * \param reconnect: automatically reconnect socket when closed. Default: true.
-     * \param priority: priority of the started task. See: Os::Task::start. Default: -1, not prioritized
-     * \param stack: stack size provided to the task. See: Os::Task::start. Default: -1, posix threads default
-     * \param cpuAffinity: cpu affinity provided to task. See: Os::Task::start. Default: -1, don't care
+     * \param priority: priority of the started task. See: Os::Task::start. Default: TASK_DEFAULT, not prioritized
+     * \param stack: stack size provided to the task. See: Os::Task::start. Default: TASK_DEFAULT, posix threads default
+     * \param cpuAffinity: cpu affinity provided to task. See: Os::Task::start. Default: TASK_DEFAULT, don't care
      */
     void startSocketTask(const Fw::StringBase &name,
                          const bool reconnect = true,
-                         const NATIVE_INT_TYPE priority = -1,
-                         const NATIVE_INT_TYPE stack = -1,
-                         const NATIVE_INT_TYPE cpuAffinity = -1);
+                         const NATIVE_UINT_TYPE priority = Os::Task::TASK_DEFAULT,
+                         const NATIVE_UINT_TYPE stack = Os::Task::TASK_DEFAULT,
+                         const NATIVE_UINT_TYPE cpuAffinity = Os::Task::TASK_DEFAULT);
 
     /**
      * \brief open the socket for communications

